src/sirinternal.c

```diff
@@ -1179,10 +1179,16 @@ pid_t _sir_gettid(void) {
     tid = get_pthread_thread_id(pthread_self());
 #elif defined(__serenity__)
     tid = gettid();
-#elif defined(_DEFAULT_SOURCE)
+#elif defined(__linux__) && defined(__GLIBC__)
+# if (__GLIBC__ >= 2 && __GLIBC_MINOR__ < 30)
```

Comment:
I think it's safe to just remove it.  I didn't because I didn't want to change the semantics, but it was for `_DEFAULT_SOURCE` (for syscall) originally, for any OS and not just Linux.  Since we handle all supported variants here and new and old glibc on Linux here, just removing is OK

Reply:
Actually need to handle the non-glibc case.  Let's see …

Reply:
@aremmell OK, this is better and even simpler.